### ISSUE TITLE
SystemSuite test for System.nanotime fails on X86

### ISSUE THREAD

**Issue by LeeTibbert** [scala-native/scala-native#458]
Yeah for unit tests. Good on the person/people who wrote them.

Another one for the X86 backlog.

This is the second  & last of two X86 test suite failures that I have
encountered to date. After my local fixs, all tests run. This
is not to say that all X86 problems have been found.
I am well aware that X86 issues may arise as test cases are added.

### Test failure
The unit-tests/src/main/scala/java/lang/SystemSuite
 test("System.nanoTime is monotonically increasing")
fao;s because, on X86, scala_native_nanotime returns
the same time in successive calls. I believe that time
is 0, but I may have become lost in type conversion
& comparison issues whilst debugging.

### Suggested fix
The suggested fix changes three files. These
changes should be safe for 64 bit systems and
even improve type correctness. 

The fix is harder to describe than to do, once
one knows files & locations.

* In  nativelib/src/main/resources/time.c, function
scala-native-nanotime change "long" to "long long" in
two(!) places: the function signature return & the
local variable nano_time.
```
long long scalanative_nano_time() {
        long long nano_time;
```
```  
   "long long" in C should be 64 bits on both 32 & 64 bit
   architectures.  I much prefer explicit types, such as
   int64_ from inttypes.h. This code compiles on
   the scala-native user's system, so one probably
   can not rely upon that header file being present.

   This change is probably the effective fix, but 
   I describe two more changes so that the types
  flow.
```
 
 *   In javalib/src/main/scala/java/lang/System.scala,
     change return type of System.nanotime() to CLongLong.
```
//  def nanoTime(): CLong = time.scalanative_nano_time                        
   def nanoTime(): CLongLong = time.scalanative_nano_time
```
```
Some magic that I do not rightly understand is happening here.

CLongLong currently resolves to type WORD, which, on 32 bit
machines is 32 bits, not 64.  This edit should be safe for 64 bit
machines, where WORD is 64 bit.

My  private test cases show that the full 64 bits are being passed
on my 32 bit machine.  With the suggested fixes in this Issue,
I get the within-human-reaction-time same values from System.nanotime
and a C program.    I never quite trust magic that I do not understand.
It usually breaks at the worst possible time.
```

  * In   nativelib/src/main/scala/scala/scalanative/runtime/time.scala,      
    change the return type of scala_native_nanotime to CLongLong.
```
 def scalanative_nano_time: CLongLong = extern
```
Q.E.D.

**Comment by LeeTibbert** [comment]
When making the fixes, please not also the suggested
integer improvement in #459.  To reduce cost,
It could be done in the same PR.

**Comment by LeeTibbert** [comment]
Closing this because it appears fixed by PR #549. Thank you @brad-rathke. 
I will also close #459, for same reason.

I tested on 32 bit X86 using both SystemSuite.scala and a manual check against a C clock_gettime() implementation. Both passed within the precision of imprecise but still useful tests.
